ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js

```diff
@@ -183,9 +183,9 @@ class Dropdown extends React.Component {
   }
   render() {
     const anchor = this.props.anchor ? (
-      React.cloneElement(this.props.anchor, { onClick: this.onToggle })
+      React.cloneElement(this.props.anchor, { onClick: () => this.onToggle() })
```

Comment:
I have heard that putting arrow functions in the render method is bad practice because you instantiate that method on every render. I'm not sure if there is a better way to set the onClick here.

Another approach that works is changed the logic on [this line](https://github.com/codice/ddf/blob/58b9ab3af4294dd863848e7365211ad47a031703/ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js#L144) to:
```
const open = typeof(next) === "boolean" ? next : !this.state.open
```

I'm not sure which solution is more "correct"

Reply:
I think wrapping it with a lambda should be fine for now.

Reply:
Yeah, it can be expensive resource wise if the function is large, but this should be pretty cheap unless something's doing something crazy with a passed in onToggle.